tools/test-proxy/transition-scripts/assets_gen.ps1

```diff
@@ -0,0 +1,87 @@
+# Criteria for generating the assets.json file
+# 1. AssetsRepo will be Azure/azure-sdk-assets
+# 2. AssetsRepoPrefixPath: <language>
+# 3. TagPrefix: <language>/<ServiceDirectory>|<language>/<ServiceDirectory>/<Library>
+# 4. Tag: Initially empty, nothing has yet been pushed
+# Tag: - Initially Empty
+class Assets {
+  [string]$AssetsRepo = "Azure/azure-sdk-assets"
+  [string]$AssetsRepoPrefixPath = ""
+  [string]$TagPrefix = ""
+  [string]$Tag = ""
+  Assets(
+    [string]$AssetsRepoPrefixPath,
+    [string]$TagPrefix
+  ) {
+    $this.TagPrefix = $TagPrefix
+    $this.AssetsRepoPrefixPath = $AssetsRepoPrefixPath
+  }
+}
+
+Function Test-Exe-In-Path {
+  Param([string] $ExeToLookFor)
+  if ($null -eq (Get-Command $ExeToLookFor -ErrorAction SilentlyContinue)) {
+    Write-Error "Unable to find $ExeToLookFor in your PATH"
+    exit 1
+  }
+}
+
+Function Get-Repo-Language {
+
+  $GitRepoOnDiskErr = "This script can only be called from within an azure-sdk-for-<lang> repository on disk."
+  # Git remote -v is going to give us output similar to the following
+  # origin  git@github.com:JimSuplizio/azure-sdk-for-java.git (fetch)
+  # origin  git@github.com:JimSuplizio/azure-sdk-for-java.git (push)
+  # upstream        git@github.com:Azure/azure-sdk-for-java (fetch)
+  # upstream        git@github.com:Azure/azure-sdk-for-java (push)
+  # We're really only trying to get the language from the git remote
+  Write-Host "git remote -v"
+  [array] $remotes = & git remote -v
+  foreach ($line in $remotes) {
+    Write-Host "$line"
+  }
+
+  # Git remote -v returned "fatal: not a git repository (or any of the parent directories): .git"
+  # and the list of remotes will be null
+  if (-not $remotes) {
+    Write-Error $GitRepoOnDiskErr
+    exit 1
+  }
+  $lang = $remotes[0] | ForEach-Object { if ($_ -match "azure-sdk-for-(.+).git") {
```

Comment:
I don't think the `.git` is necessarily guaranteed, e.g.

```
⇉ ⇉ ⇉ git remote -v
fork    git@github.com:benbp/azure-sdk-tools (fetch)
fork    git@github.com:benbp/azure-sdk-tools (push)
origin  git@github.com:Azure/azure-sdk-tools.git (fetch)
origin  git@github.com:Azure/azure-sdk-tools.git (push)
```

Reply:
```suggestion
  $lang = $remotes[0] | ForEach-Object { if ($_ -match "azure-sdk-for-([a-zA-Z0-9\-\_\.]+)?(\.git)?") {
```

To account for ben's feedback

Reply:
It'll be updated with the next commit